### ISSUE TITLE
No automatic lights

### ISSUE THREAD

**Issue by danielbrunt57** [jseidl/hass-magic_areas#88]
Updated to v2.0.3 but still no automatic lights for living room.
Office is working.
Bedroom, bathroom are not working.

Here is what I had before config flow:
```
  living_room:
    icon: 'mdi:sofa'
    presence_sensor_device_class: ['motion']
    automatic_lights:
      entities: 
       - light.fireplace_light
      disable_entity: binary_sensor.living_room_motion_override
    include_entities:
      - binary_sensor.lumi_sensor_motion_living_room
      - light.fireplace_light
      - light.pole_lamp
      - light.living_room_light
      - media_player.living_room_tv
```
Here is what I see now:
![image](https://user-images.githubusercontent.com/49846893/106430398-9535e380-6420-11eb-8233-66a21d8ea194.png)

Here is the config flow setup:
![image](https://user-images.githubusercontent.com/49846893/106430513-c57d8200-6420-11eb-898d-6b57583d4cd9.png)

**Comment by danielbrunt57** [comment]
Seems to be horribly FUBAR now...
I don't like `night'... should be `disable`.
I may have to revert to entity_controller...

**Comment by jseidl** [comment]
Night was created to allow other features use this "disable at night" functionality.

Basically when `night_entity` is on `night_state` the area will be in `night` mode. Automatic lights only will be enabled if `night` is `on`.

In most `disable` migration cases, the state should be the opposite of what it was. So if your `binary_sensor.living_room_motion_override` was used to disable autolights, it is now set to enable so you must invert the state you had before.

**Comment by danielbrunt57** [comment]
Ok, so I renamed the binary sensor and reversed the logic...
```
# Living Room Night Mode
      living_room_night_mode_enabled:
        unique_id: "living-room-night-mode"
        friendly_name: "Living Room Night Mode Enabled"
        value_template: "{{ is_state('binary_sensor.south_patio_lux_override', 'off') and is_state('input_boolean.living_room_automatic_lights_disabled', 'off') and is_state('light.pole_lamp', 'off') and is_state('light.living_room_light', 'off') }}"
```
magic areas yaml:
```
  living_room:
    icon: 'mdi:sofa'
    presence_sensor_device_class: ['motion']
    main_lights:
     - light.fireplace_light
    night_entity: binary_sensor.living_room_night_mode_enabled
    include_entities:
      - binary_sensor.lumi_sensor_motion_living_room
      - light.fireplace_light
      - light.pole_lamp
      - light.living_room_light
      - media_player.living_room_tv
```
Fireplace light not turning on despite presence is detected and automatic_lights still says disabled...
![image](https://user-images.githubusercontent.com/49846893/106450660-514fd800-643a-11eb-9007-1218839158b0.png)

It would appear that configuration.yaml specification for magic areas does not have any effect...have to use config flow.
I've removed the manual config for magic areas from configuration.yaml and reconfigured living room using config flow.
automatic lights now says enabled and living room light turned on when i walked into the room.

![image](https://user-images.githubusercontent.com/49846893/106451772-e7383280-643b-11eb-95c2-2821dc2d2e47.png)

I'll convert my other areas now.........

**Comment by TheArcturian** [comment]
jseidl: "Automatic lights is disabled at day (when not night mode)."

I can't get my light to turn on at daytime. When Night mode is true, it actually works for the lights defined as night light. However when it is daytime (Night = false), no lights gets turned on.

**Comment by danielbrunt57** [comment]
Success.  
Office working as before with no night mode (24x7 presence detection)
Living room working as before with night mode & alternate lights/TV as overrides
Front Porch working as before (using new night  mode sensor which is the reverse of my original outside lux override that's on during daytime)
Bedroom working as before with night mode except when sleep mode is turned on.
Bathroom working as before with no night mode (24x7 motion activated).

**Comment by TheArcturian** [comment]
What exactly was the solution?

**Comment by danielbrunt57** [comment]
@TheArcturian You need to leave `Entity used to track night state` blank if you want 24x7 activation of the light(s)...

**Comment by danielbrunt57** [comment]
I had to reverse my version 1.x logic for the new night state.  Foe night time operation the entity specified has to be true.  If the entity is not specified (blank) then automatic lights is enabled 24x7....

**Comment by TheArcturian** [comment]
I noticed that the rooms not having that one set was the areas working. However it is impossible to remove it once it is set to a binary sensor. Were you able to remove it?

Also it should be possible to use night mode as well. I noticed it turned on the lights when Night = true, but not when Night = false.

**Comment by danielbrunt57** [comment]
Yes.  Go to the top of the list and select the blank line at the top!

**Comment by danielbrunt57** [comment]
> Also it should be possible to use night mode as well. I noticed it turned on the lights when Night = true, but not when Night = false. This must still be a bug.

That's not a bug!  If you WANT to use night mode for an area AND have 24X7 operation then that night mode entity will have to ALWAYS be true....  Just remove the night mode entity!

**Comment by danielbrunt57** [comment]
HUGE breaking change from ver 1.x, and not well documented!  
There will be LOTS of confused users!!

**Comment by danielbrunt57** [comment]
4:30 am here...time for a little shut eye now before work in 4 hours!

**Comment by TheArcturian** [comment]
Thanks for your feedbacks. I still am not able to save by selecting the blank line. Did you go back and check if it actually was saved correctly? No matter what browser or platform I'm using, it is impossible for me to save the blank line at the top.

**Comment by jseidl** [comment]
> HUGE breaking change from ver 1.x, and not well documented!
There will be LOTS of confused users!!

There is a lot of breaking changes, that's why I waited so much to do those changes and do a major version release. I do expect to have a few bugs lingering and your feedback is of utmost importance on solving them. Thanks for all the report.

I definitely need to write a *Migration guide* for the existing users. 

Thanks for reporting the YAML issue as well, it's hard to make both (yaml and config flow) to go hand in hand, I'll work on fixing that.

> I'm really confused by setting Night = true to get daytime lights to go on. 

You have to have no night entities if you want the lights in the room to go on during daytime. There is a bug in the UI that won't let you clear the selection if you added one. I'm working on this. Meanwhile you can delete the area from the integrations menu and restart, area should be re-added with the default settings.

**Comment by danielbrunt57** [comment]
> You have to have no night entities if you want the lights in the room to go on during daytime. There is a bug in the UI that won't let you clear the selection if you added one. I'm working on this. Meanwhile you can delete the area from the integrations menu and restart, area should be re-added with the default settings.

You are correct...going back and saving the blank entity does NOT erase the field!  What I observed only works within the present editing session.  Once it's saved, it can't be undone other than deleting/re-adding the integration for the area.

> I'm really confused by setting Night = true to get daytime lights to go on. I do want to disable the lights to go on during the night, so making that work, I have to set my binary_sensor.night_mode to On so "Night = true"?

I don't have a scenario exactly like that here.  Closest would be my bedroom where I use night_entity but the sensor turns off when I turn on another entity sleep_mode. If you want automatic lights during the daytime only then I would say use night_entity but reverse the logic of the entity specified (call it binary_sensor.day_mode?) so the entity is on during the day but off at night.

**Comment by jseidl** [comment]
Created #94 to track the new documentation updates that is required to clear the confusion and clarify the breaking changes. Thanks for all your input folks.

**Comment by danielbrunt57** [comment]
> Created #94 to track the new documentation updates that is required to clear the confusion and clarify the breaking changes and #95 to track the dropdown issue.
> 
> Thanks for all your input folks.

NP!

**Comment by TheArcturian** [comment]
> 2. Use your regular night_entity but for that area, change the state required from 'on' to 'off' so that the logic test results in 'true'...
>    ![image](https://user-images.githubusercontent.com/49846893/106502375-246ee580-6479-11eb-9291-a2d66aeea81b.png)
>    If `off` is specified, during the daytime the logic would be `if night_entity = 'off'` and would return true in the daytime.

Yeah, I did change that setting to Off so that daytime lights can go on. I'll see if it works as expected when nighttime comes. :)

**Comment by jseidl** [comment]
@TheArcturian I've posted a diagram on #94 that shall make things clearer

**Comment by danielbrunt57** [comment]
> @TheArcturian I've posted a diagram on #94 that shall make things clearer

Excellent work!

**Comment by danielbrunt57** [comment]
After viewing those flowcharts I am wondering if sleep_entity could be modified such that if permits usage without specifying an entity to be controlled, essentially resulting in a No OP?

![image](https://user-images.githubusercontent.com/49846893/106537165-a3cadc00-64ae-11eb-917c-fa0ba34b18d2.png)

I do not have any sleep lights myself but I could make use of the sleep mode and use the entity I am already using for Circadian Lighting: `input_boolean.sleep_mode`.  I just don't have any lights I would want it to turn on..

I could also use it for my fireplace light scenario.  Different input_boolean I would turn on if living room light, pole lamp or TV are turned on...

**Comment by jseidl** [comment]
I do use circadian_lighting (adaptive_lighting) sleep switch as my `sleep_entity`. What's the use case for having an area in sleep without different set of lights to control?

**Comment by danielbrunt57** [comment]
I don't have any sleep lights, yet!
For testing, I created an input_boolean and then templated a light for it.  I can now specify that dummy light for my living room sleep light.  Changed night mode to be based just on outside lux level template.  living_room_motion_override is based on any of three entities being on and that is specified as my sleep entity.  The area switches to sleep mode when one of those three entities turns on,  While in sleep mode and when motion is detected, it turns on the dummy sleep light.  What I observed about that behavior though is that if motion is detected while sleep mode is off, it turns the fireplace light on but if sleep mode is turned on while in that state, it does not switch from fireplace light to sleep light.  When area is clear, it turns off the fireplace light and when area presence is detected again (motion), it turned the dummy sleep light on.  

Maybe, I can change the fireplace light to be sleep light and sleep mode for this area turns on at sunset, off at sunrise.  Would still need to template an entity for 2 lights + TV that would disable sleep mode but then do I need an entity for it to control when not in sleep mode??  Hmmmm....

Basically, the fireplace light is a night light triggered by motion that illuminates living room and hallway that leads to kitchen.
No need for it turn on if living room light or pole lamp are on so specifying other lights or TV as presence sensors does not work since MA would turn on the fireplace light.  If TV is on then most likely one of the three lights is on and if only the fireplace light is on I don't want it turning off because motion has ceased...  If you find this hard to follow, no worries!

**Comment by jseidl** [comment]
When I have complex situations I try to rethink if I need that complexity at all or if there's another way to reach similar results. With Magic Area I try to steer away from the concept as motion being the trigger and move towards presence being the trigger. 

The issue that other lights are a condition for the state of the fireplace light, leads me to believe maybe the problem is being approached wrong (in the universe of Magic Areas).

Given that, there are some questions:

* When Living Room is occupied, would you like to lights come up all the time or only at night? 
* When Living Room is occupied, which lights you ALWAYS want to turn on? (What are your main lights)
* When Living Room is occupied deep into the night (i.e. sleep), which lights you want to turn on? (If any - What are your sleep lights)
* Which area benefits from Fireplace Light being on? (Living Room, Hallway?)

Let's lay down some situations given a scenario where you have `light.living_room_light` as your Main Light and Fireplace Light as your sleep light. Pole lamp is not turned on by MA in any situation (only turned off when area is clear).

* You're just passing by the Living Room, looking for your wallet, room is marked as Occupied
** It's `day`, no lights comes on.
** It's `night`, your main light comes on.
** It's `sleep` (late at night), your fireplace light comes on instead of the main light.
* You wanna read. You walk into the Living Room, room is marked as Occupied, (same behavior as above). You sit down, turn on your pole light.

Then there's a case for a separate automation. In my living room I like to turn off the ceiling lights and turn on the accent lights (hue play) on the sides of the tv rack. I the TV being on as a presence sensor, so while I'm watching it it won't clear the area and turn the lights off. I have a separate automation that when TV is on, do this light change and when goes off, reverse.

**Comment by danielbrunt57** [comment]
I agree about trying to rethink my approach.  For your questions:
When Living Room is occupied, would you like to lights come up all the time or only at night?
 - only at night
When Living Room is occupied, which lights you ALWAYS want to turn on? (What are your main lights)
 - Usually fireplace light
When Living Room is occupied deep into the night (i.e. sleep), which lights you want to turn on?
 - Fireplace light
Which area benefits from Fireplace Light being on?
  For nighttime illumination purposes, it's really a small area (<700 sq ft) so the majority of the main floor...
 1. Living Room
 2. Hallway, but not really a hallway in traditional terms; it's just a straight through from the landing up from the front door rather a right turn to the living room, past a coat closet on right and powder room on left into the kitchen.
 3. Front entrance (1/2 level down)
 3. Office
 4. Kitchen

 > I have a separate automation that when TV is on, do this light change and when goes off, reverse.

I actually thought about that and threw in an automation to do just that.  After some testing though, I ended up changing MA for the living room to be just presence detection rather than for light control.  

![image](https://user-images.githubusercontent.com/49846893/106559101-c7574c00-64d9-11eb-9cb2-753f3e23f807.png)

and I now have this automation:
```
alias: Automatic Lights - Living Room
description: ''
trigger:
  - platform: state
    entity_id: light.pole_lamp
  - platform: state
    entity_id: light.living_room_light
  - platform: state
    entity_id: light.fireplace_light
  - platform: state
    entity_id: media_player.living_room_tv
  - platform: state
    entity_id: binary_sensor.area_living_room
condition:
  - condition: state
    entity_id: binary_sensor.south_patio_lux_override
    state: 'off'
action:
  - choose:
      - conditions:
          - condition: and
            conditions:
              - condition: state
                entity_id: light.pole_lamp
                state: 'off'
              - condition: state
                entity_id: light.living_room_light
                state: 'off'
              - condition: state
                entity_id: light.fireplace_light
                state: 'off'
              - condition: or
                conditions:
                - condition: state
                  entity_id: media_player.living_room_tv
                  state: 'on'
                - condition: state
                  entity_id: binary_sensor.motion_living_room
                  state: 'on'
        sequence:
          - service: light.turn_on
            data: {}
            entity_id: light.fireplace_light
      - conditions:
          - condition: state
            entity_id: light.pole_lamp
            state: 'off'
          - condition: state
            entity_id: light.living_room_light
            state: 'off'
          - condition: state
            entity_id: light.fireplace_light
            state: 'on'
          - condition: state
            entity_id: media_player.living_room_tv
            state: 'off'
          - condition: state
            entity_id: binary_sensor.area_living_room
            state: 'off'
        sequence:
          - service: light.turn_off
            data: {}
            entity_id: light.fireplace_light
      - conditions:
          - condition: or
            conditions:
              - condition: state
                entity_id: light.pole_lamp
                state: 'on'
              - condition: state
                entity_id: light.living_room_light
                state: 'on'
          - condition: state
            entity_id: binary_sensor.area_living_room
            state: 'on'
        sequence:
          - service: light.turn_off
            data: {}
            entity_id: light.fireplace_light
    default: []
mode: single
```
I thing I might have what I want with an automation using MA presence detection rather than complex templating and trying to get MA to control the lights..

**Comment by jseidl** [comment]
Indeed brother, `control_lights` is aimed more towards this general use. On these situations you're better off managing with a dedicated automation. Those new `choose` actions are dope, ended up using for my front door that has a non-standard light control routine.

**Comment by andyinno** [comment]
I think that some confusion could come from the naming of the states. At least for me.

Currently it is defined a "night state".
If the night state is defined and the night state is off, then it goes to "no op"

In my case for example I use xiaomi motion sensor in order to define if an area is occupied or not, and I use the illumination level that are provided by the xiaomi motion sensor in order to define if the room is too bright for turning on the light.

@jsedl use an external sensor in order to define if it is night and day, I use internal sensors (I have shutters on Windows, when I leave home shutters closes) when I come back at home, in full day, the room is dark, while shutters are opening I "need" lights on.

I need to use night state as "dark state" meaning: the room is dark, turn on the lights.
For my workflow dark state would be a better naming for the state and the sensor.

what do you think?

**Comment by danielbrunt57** [comment]
Personally, I would define the state more generically as `enabled_state` with an `enable_entity` and `enable_mode` and get way from using `Night` or 'Dark' but that's just my opinion...

Entity used to control area state: 
State of enable entity:

Then in the area attributes
Automatic  [enabled|disabled] rather than `Automatic` **`lights`** since other features can be controlled (climate, etc.)
Enabled  [true|false] rather than `Night`
Sleep  [true|false]

**Comment by andyinno** [comment]
makes sense. more general

**Comment by danielbrunt57** [comment]
(see edits to my post above)

**Comment by jseidl** [comment]
You both have valid comments but you're looking only into the Automatic Lights optic. These states were chosen as night and sleep to accommodate other features, for example, area-aware media player will not broadcast notifications for areas sleeping, by default.

Different features benefit from these states in different ways. Most notably is Automatic Lights of course but it's not restricted to that.

**Comment by danielbrunt57** [comment]
I agree with keeping sleep but could sleep at least be changed so that an entity is optional rather than required?

With regards to `night`, I tend to disagree with you. In keeping with the concept of generality I feel `enabled` or `automatic` is more general than `night`.  Night implies it is dark and you want more light but to override the automatic lighting based on presence detection I need to turn off `night` mode. For me turning off an `automatic` mode makes more sense...

**Comment by jseidl** [comment]
> I agree with keeping sleep but could sleep at least be changed so that an entity is optional rather than required?

Yeah that's legacy from when it was tied to lights, will be changed.

> Night implies it is dark and you want more light but to override the automatic lighting based on presence detection I need to turn off night mode. For me turning off an automatic mode makes more sense...

Maybe it would more sense instead of defining `night`, one defined `day`? This way you would be setting a `disable` entity.

**Comment by danielbrunt57** [comment]
> Yeah that's legacy from when it was tied to lights, will be changed.

That's good news~



> Maybe it would more sense instead of defining `night`, one defined `day`? This way you would be setting a `disable` entity.

I still think for more general use it and get away from being tied to lighting, `automatic` or `motion_control` or something like that is easier to think about.
```
automatic_mode: enabled | disabled
automatic_control_entity: binary_sensor.xyz
  control_state: on (means area entities will be controlled via presence detection when automatic_control_entity is **on**)
  control_state: off (logic reversed - means area entities will be controlled via presence detection when automatic_control_entity is **off**
```
I presume you've tried/used [Daniel Mason's Entity Controller](https://github.com/danobot/entity-controller) before?

**Comment by jseidl** [comment]
I did and I believe Entity Controller is the more general use focused component. Magic Areas has area state and presence detection as main focus and actions are features, so features are aimed towards are state rather than being more generic.
